pkg/controller/datavolume/util.go

```diff
@@ -292,7 +292,7 @@ func resolveVolumeSize(c client.Client, dvSpec cdiv1.DataVolumeSpec, pvcSpec *v1
 
 	if !found {
 		// Storage size can be empty when cloning
-		isClone := dvSpec.Source.PVC != nil
+		isClone := dvSpec.Source.PVC != nil || dvSpec.Source.Snapshot != nil
```

Comment:
We don't do regular size-detection when cloning from a snapshot, are we sure this works? For example, what would happen if the target DV size is empty and the snapshot `restoreSize` is zero?

Reply:
covered by https://github.com/kubevirt/containerized-data-importer/blob/5f85c423e16c71de32d3011eaca2545afb8c347d/pkg/controller/common/util.go#L790-L792